### DISCUSSION TITLE
how to get the covariance matrix out of the CovarianceEstimator class?

### DISCUSSION THREAD

**Discussion by nemoincognito** [skfolio/skfolio#52] [Q&A] [answered]
First of all - amazing project, really fantastic stuff. 

I was reading this section: https://skfolio.org/generated/skfolio.moments.DenoiseCovariance.html#skfolio.moments.DenoiseCovariance

and tried to extract the covariance matrix from that estimator using:

`model = EmpiricalPrior(
    mu_estimator=ShrunkMu(method=ShrunkMuMethods.BODNAR_OKHRIN, covariance_estimator=DenoiseCovariance()))
model.fit(X)
model.covariance_estimator_`

Then tried to find it using 
`dir(model)`

But cannot find the ndarray that has the covariance matrix. How do I extract this? 

I am keen to compare different estimation approaches and also look at different sampling windows and periods for the covariance matrix in a couple of Black-Litterman implementations.

**Comment by HugoDelatte** [comment, marked as answer]
Thank you!

Here you go:

```python
from skfolio.datasets import load_sp500_dataset
from skfolio.preprocessing import prices_to_returns
from skfolio.prior import EmpiricalPrior
from skfolio.moments import ShrunkMu, ShrunkMuMethods, DenoiseCovariance

prices = load_sp500_dataset()

X = prices_to_returns(prices)

model = EmpiricalPrior(
    mu_estimator=ShrunkMu(method=ShrunkMuMethods.BODNAR_OKHRIN),
    covariance_estimator=DenoiseCovariance()
)

model.fit(X)

print(model.prior_model_.covariance)
```

The user guide for the prior model can be found here: https://skfolio.org/user_guide/prior.html

But to simply compare covariance estimators, you can do:

```python
model = DenoiseCovariance()
model.fit(X)
print(model.covariance_)
```

**Reply by nemoincognito** [reply]
Thanks very much.
